scene/2d/particles_2d.cpp

```diff
@@ -707,6 +713,18 @@ bool Particles2D::is_emitting() const {
 	return emitting;
 }
 
+void Particles2D::set_process_mode(ProcessMode p_mode) {
+
+	process_mode=p_mode;
+	set_process(emitting && process_mode==PROCESS_IDLE);
+	set_fixed_process(emitting && process_mode==PROCESS_FIXED);
```

Comment:
What if it isn't emitting but there are still particles alive (amount > 0)? Wouldn't you just freeze everything then?